pkg/test/runner/runner.go

```diff
@@ -246,6 +247,18 @@ func (r *Runner) runFnEval() error {
 	return nil
 }
 
+// Match strings starting with [PASS] or [FAIL] and ending with " in N...". We capture the duration portion
+var timestampRegex = regexp.MustCompile(`\[(?:PASS|FAIL)].* in ([0-9].*)`)
+
+func sanitizeTimestamps(stderr string) string {
+	// Output will have non-deterministic output timestamps. We will replace these to static message for
+	// stable comparison in tests.
+	for _, m := range timestampRegex.FindAllStringSubmatch(stderr, -1) {
+		stderr = strings.ReplaceAll(stderr, m[1], "0s")
```

Comment:
This introduces friction in adding tests. Today, users just copy paste the output in the expected `stderr` section, not they need to replace timing with `0s` after copy-paste. Not a huge deal given the benefits, but don't have better solution in mind also.

/cc @mengqiy this will have ripple effects on function-catalog tests when we update the test runner in that repo.

Reply:
we could make `KPT_E2E_UPDATE_EXPECTED` (or a new var) auto update it if desired?

Reply:
Or also sanitize the expected string as well. That may mean that there is a lot of churn in the golden file though

Reply:
I think adding a new var (KPT_E2E_UPDATE_EXPECTED_STDERR_TIMING) for auto-updating the expected stderr with `0ms` can work well. So auto-update behavior is to replace any timing info with `0s` or add `0s` if timing info is missing in the expected stderr (this can help minimize the ripple effect in the function-catalog-repo).

Reply:
> this will have ripple effects on function-catalog tests when we update the test runner in that repo.

We are NOT testing presentation layer in the catalog. Instead we test it through the structured result API. So this change is not a problem for the catalog repo.

IIRC we have a few test in the kpt repo that are testing the presentation layer. They will be affected.

Reply:
ok, that's a fair point as well.